org.adempiere.base/src/org/compiere/model/MCost.java

```diff
@@ -1437,10 +1541,19 @@ public static MCost get (Properties ctx, int AD_Client_ID, int AD_Org_ID, int M_
 		final Object[] params = new Object[]{AD_Client_ID, AD_Org_ID, M_Product_ID,
 												M_CostType_ID, C_AcctSchema_ID,
 												M_CostElement_ID, M_AttributeSetInstance_ID};
-		return new Query(ctx, Table_Name, whereClause, trxName)
+		MCost cost = new Query(ctx, Table_Name, whereClause, trxName)
 					.setOnlyActiveRecords(true)
 					.setParameters(params)
 					.firstOnly();
+		// use the quantity, cost price, accumulated qty and accumulated amt from the cost history record
+		if (history != null)
+		{
+			cost.setCurrentQty(history.getNewQty());
+			cost.setCurrentCostPrice(history.getNewCostPrice());
+			cost.setCumulatedQty(history.getNewCQty());
+			cost.setCumulatedAmt(history.getNewCAmt());

```

Comment:
This looks abit strange, why the need to query and get MCost record just to replace with details from MCostHistory ?

Reply:
When posting a back-date transaction or re-posting an existing transaction, we want to get the costing information (i.e. CurrentQty, CurrentCostPrice, CumulatedQty & CumulatedAmt) at that point (from MCostHistory). If there is no MCostHistory found, then use the costing information from the current MCost.